fenix/app/src/main/java/org/mozilla/FenixActivity.kt

```diff
@@ -0,0 +1,13 @@
+package org.mozilla
+
+import androidx.navigation.NavDestination
+import androidx.navigation.NavDirections
+import mozilla.components.support.utils.SafeIntent
+import org.mozilla.fenix.BrowserDirection
+
+interface FenixActivity {
+    fun getBreadcrumbMessage(destination: NavDestination): String
+    fun getIntentSource(intent: SafeIntent): String?
+    fun getIntentSessionId(intent: SafeIntent): String?
+    fun getNavDirections(from: BrowserDirection, customTabSessionId: String?): NavDirections?
```

Comment:
Is there a plan for how these functions will be called from `ExternalAppBrowserActivity`? It looks like this was the main point of the inheritance - AFAICT these functions are only called from within `HomeActivity`, which means we would be missing that inherited functionality in `ExternalAppBrowserActivity` unless we explicitly replicate it. For example, `ExternalAppBrowserActivity` used to inherit this line from HomeActivity and only overwrite the `::getBreadcrumbMessage`. 

```
// in home activity
lifecycle.addObserver(
    BreadcrumbsRecorder(
        components.analytics.crashReporter,
        navHost.navController,
        ::getBreadcrumbMessage,
    ),
)
```

Reply:
Unless I'm misunderstanding the comment, [these functions](https://github.com/mozilla-mobile/firefox-android/blob/825175e8acaddd47c1899ea5169c74049ce61043/fenix/app/src/main/java/org/mozilla/fenix/customtabs/ExternalAppBrowserActivity.kt#L48)  were already being overridden in `ExternalAppBrowserActivity` but they haven't had to change in the file as it's still using those existing functions but getting them from `FenixActivity` instead.

Reply:
Ah, I was assuming the step of removing the `HomeActivity` inheritance from `ExternalAppBrowserActivity`. I didn't see that the example was still doing that 😅